node/src/proxy_server/mod.rs

```diff
@@ -4157,6 +4732,367 @@ mod tests {
+        )];
+        let route_query_response_expected = RouteQueryResponse {
+            route: make_meaningless_route(),
+            expected_services: ExpectedServices::RoundTrip(
+                expected_services.clone(),
+                expected_services.clone(),
+                1234,
+            ),
+        };
+        let neighborhood_mock = neighborhood_mock
+            .stop_on_message(type_id!(RouteQueryMessage))
+            .route_query_response(Some(route_query_response_expected.clone()));
+        let cryptde = main_cryptde();
+        let mut subject = ProxyServer::new(
+            cryptde,
+            alias_cryptde(),
+            true,
+            Some(STANDARD_CONSUMING_WALLET_BALANCE),
+            false,
+        );
+        let socket_addr = SocketAddr::from_str("1.2.3.4:5678").unwrap();
+        let mut dns_failure_retries_hash_map = HashMap::new();
+        let client_payload = make_request_payload(111, cryptde);
+        let stream_key = client_payload.stream_key;
+        dns_failure_retries_hash_map.insert(
+            stream_key,
+            DNSFailureRetry {
+                unsuccessful_request: client_payload.clone(),
+                retries_left: 3,
+            },
+        );
+        subject.dns_failure_retries = dns_failure_retries_hash_map;
+        subject
+            .keys_and_addrs
+            .insert(stream_key.clone(), socket_addr.clone());
+        subject.route_ids_to_return_routes.insert(
+            1234,
+            AddReturnRouteMessage {
+                return_route_id: 1234,
+                expected_services: expected_services.clone(),
+                protocol: ProxyProtocol::HTTP,
+                hostname_opt: Some("server.com".to_string()),
+            },
+        );
+        let message_resolver = RouteQueryResponseResolverMock::default()
+            .resolve_message_params(&resolve_message_params_arc);
+        let message_resolver_factory = RouteQueryResponseResolverFactoryMock::default()
+            .make_result(Box::new(message_resolver));
+        subject.inbound_client_data_helper_opt = Some(Box::new(IBCDHelperReal {
+            factory: Box::new(message_resolver_factory),
+        }));
+        let subject_addr: Addr<ProxyServer> = subject.start();
+        let dns_resolve_failure = DnsResolveFailure_0v1::new(stream_key);
+        let expired_cores_package: ExpiredCoresPackage<DnsResolveFailure_0v1> =
+            ExpiredCoresPackage::new(
+                SocketAddr::from_str("1.2.3.4:1234").unwrap(),
+                Some(make_wallet("irrelevant")),
+                return_route_with_id(cryptde, 1234),
+                dns_resolve_failure.into(),
+                0,
+            );
+        let mut peer_actors = peer_actors_builder()
+            .neighborhood(neighborhood_mock)
+            .build();
+        peer_actors.proxy_server = ProxyServer::make_subs_from(&subject_addr);
+
+        subject_addr.try_send(BindMessage { peer_actors }).unwrap();
```

Comment:
`BindMessage` isn't part of the Act section; it should be clustered up with the Arrange part.